hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/GcsEventsHoodieIncrSource.java

```diff
@@ -168,10 +182,18 @@ public Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastChec
           + queryInfo.getStartInstant());
       return Pair.of(Option.empty(), queryInfo.getStartInstant());
     }
+    return fetchPartitionedSource(cloudObjectIncrCheckpoint, queryRunner.run(queryInfo, snapshotLoadQuerySplitter), sourceLimit);
+  }
 
-    Pair<QueryInfo, Dataset<Row>> queryInfoDatasetPair = queryRunner.run(queryInfo, snapshotLoadQuerySplitter);
+  private Pair<Option<Dataset<Row>>, String> fetchPartitionedSource(CloudObjectIncrCheckpoint cloudObjectIncrCheckpoint, Pair<QueryInfo, Dataset<Row>> queryInfoDatasetPair, long sourceLimit) {
+    if (this.sourceProfileSupplier.isPresent() && this.sourceProfileSupplier.get().getSourceProfile() != null) {
```

Comment:
can we introduce a boolean variable `isSourceProfileSupplierAvailable` and use it everywhere